internal/config/types.go

```diff
@@ -103,15 +96,31 @@ type (
 	Processor struct {
 		Type string `yaml:"-" mapstructure:"type"`
 	}
-
 	// OTel Collector Receiver configuration.
-	Receiver struct {
+	Receivers struct {
+		OtlpReceivers      []OtlpReceiver      `yaml:"-" mapstructure:"otlp_receivers"`
+		NginxReceivers     []NginxReceiver     `yaml:"-" mapstructure:"nginx_receivers"`
+		NginxPlusReceivers []NginxPlusReceiver `yaml:"-" mapstructure:"nginx_plus_receivers"`
+		HostMetrics        bool                `yaml:"-" mapstructure:"host_metrics"`
```

Comment:
consistency: either we should rename `HostMetrics` to `HostMetricsReceivers` or remove the `Receivers` suffix from all of these altogether. I would opt for the latter approach, as the suffix is redundant because we are within the `Receivers` struct.